sqlmesh/core/analytics/collector.py

```diff
@@ -0,0 +1,351 @@
+        """
+        self._on_command("PYTHON_API_COMMAND", command_name, command_args)
+
+    def on_project_loaded(
+        self,
+        *,
+        project_type: str,
+        models_count: int,
+        audits_count: int,
+        standalone_audits_count: int,
+        macros_count: int,
+        jinja_macros_count: int,
+        load_time_sec: float,
+        state_sync_fingerprint: str,
+        project_name: str,
+    ) -> None:
+        """Called when a project is loaded.
+
+        Args:
+            project_type: The type of the project. Eg. "dbt" or "native".
+            models_count: The number of models in the project.
+            audits_count: The number of audits in the project.
+            standalone_audits_count: The number of standalone audits in the project.
+            macros_count: The number of macros in the project.
+            jinja_macros_count: The number of Jinja macros in the project.
+            load_time_sec: The time it took to load the project in (fractional) seconds.
+            state_sync_fingerprint: The fingerprint of the state sync configuration.
+            project_name: The name of the project.
+        """
+        dbt_version = None
+        project_type = project_type.upper()
+        if project_type == "DBT":
+            from dbt import version
+
+            dbt_version = version.__version__
+
+        self._add_event(
+            "PROJECT_LOADED",
+            {
+                "project_type": project_type.upper(),
+                "models_count": models_count,
+                "audits_count": audits_count,
+                "standalone_audits_count": standalone_audits_count,
+                "macros_count": macros_count,
+                "jinja_macros_count": jinja_macros_count,
+                "load_time_ms": int(load_time_sec * 1000),
+                "state_sync_fingerprint": state_sync_fingerprint,
+                "project_name_hash": _anonymize(project_name),
+                "dbt_version": dbt_version,
+            },
+        )
+
+    def on_plan_apply_start(
+        self,
+        *,
+        plan: Plan,
+        engine_type: t.Optional[str],
+        state_sync_type: t.Optional[str],
+        scheduler_type: str,
+    ) -> None:
+        """Called after the plan application starts.
+
+        Args:
+            plan: The plan that is being applied.
+            engine_type: The type of the target engine.
+            state_sync_type: The type of the engine used to store the SQLMesh state.
+            scheduler_type: The type of the scheduler being used. Eg. "builtin" or "airflow".
+        """
+        self._add_event(
+            "PLAN_APPLY_START",
+            {
+                "plan_id": plan.plan_id,
+                "engine_type": engine_type.lower() if engine_type is not None else None,
+                "state_sync_type": state_sync_type.lower() if state_sync_type is not None else None,
+                "scheduler_type": scheduler_type.upper(),
```

Comment:
let's make everything lower

Reply:
Why? Isn't convention for enum values to be upper case?

Reply:
maybe? but you make engine_type and state_sync_type lower, so now it's inconsistent, additionaly, things like dbt, native, sqlmesh etc, are all normally represented as lower case